cmd/dagger/module.go

```diff
@@ -380,9 +380,11 @@ If not updating source or SDK, this is only required for IDE auto-completion/LSP
 				return fmt.Errorf("failed to generate code: %w", err)
 			}
 
-			// If no license has been created yet, we should create one.
-			if err := findOrCreateLicense(ctx, modConf.LocalRootSourcePath); err != nil {
-				return err
+			// If no license has been created yet, and SDK is set, we should create one.
+			if developSDK != "" {
```

Comment:
I suggest setting the default value for the flag to be `defaultLicense` (in `licenses.go`), instead of implicitly using the default on an empty (`""`) value. This way you can check for empty license here as well, and not create it if empty:

```suggestion
			if developSDK != "" && licenseID != "" {
```

This way it's clearer when looking at `--help` what the default is, and it also allows to opt-out of creating the file (`--license ""`) as today it's not possible.

Note that there's another bug that you can fix right now. The `--license` flag is not defined on `dagger develop` like it is in `dagger init`. So in fact users can't choose any license atm.

Reply:
> (--license "") as today it's not possible. Yes but that means if users already has a LICENSE at the top dir, it's going to create a license too if it's not set to `""`.
Should I also update this part of the code to not create a license no matter what if it already exist in the repo?

Reply:
Of course, rather than check for `""` you check if it's the default or not. That preserves current behavior in other parts of the code.

Reply:
But what if the user's want the license `Apache-2.0` ?? and set `license="Apache-2.0"`, it's not going to work as expected and will not create it if I test against the default.

Reply:
I'm going the push as you requested but I'm not sure that's the right solution, instead if would be better to check if the flag has been explicitely set so we know if we generate a license (like in #7715)

Reply:
> But what if the user wants the license `Apache-2.0` ?? and set `license="Apache-2.0"`, it's not going to work as expected and will not create it if I test against the default.

You can check if the flag is set or not. You don't have to depend on the value alone.

Reply:
Afaict, if you explicitly set `--license=Apache-2.0` with a license in a parent directory, it wouldn't create it. While if the value is different from the default such as `--license=MIT` then it would. So I added a check to search parent dirs only if flag not set, independently of the value.